api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.expressions.partitions.Partition;
+import java.util.Map;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
+
+  /**
+   * Get a partition by partition name
+   *
+   * @param partitionName the name of the partition
+   * @return the partition
+   * @throws NoSuchPartitionException if the partition does not exist
+   */
+  Partition getPartition(String partitionName) throws NoSuchPartitionException;
+
+  /**
+   * Check if a partition exists.
+   *
+   * @param partitionName The name of the partition.
+   * @return True if the partition exists, false otherwise.
+   */
+  default boolean partitionExists(String partitionName) {
+    try {
+      getPartition(partitionName);
+      return true;
+    } catch (NoSuchPartitionException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Add a partition with specified name and properties to the table.
+   *
+   * @param partitionName The identifier of the partition.
+   * @param properties The properties of the partition.
+   * @return The created partition.
+   * @throws PartitionAlreadyExistsException If the partition already exists.
+   */
+  Partition addPartition(String partitionName, Map<String, String> properties)
```

Comment:
Could we have more constraints for partitions? such as partition specification.

Reply:
Partitioned by which value. HashParition vs RangeParition. Primary partition vs Secondary partition.

Reply:
> Partitioned by which value

It's already defined by `Transform`.

I guess I know what you mean, I should change the interface parameters to `Partition`, right?

Reply:
Another question: Is the partition name equal to the partition value?

Reply:
No exactly, these are two different attributes of a partition object, and we should not confuse them.

Reply:
> No exactly, these are two different attributes of a partition object, and we should not confuse them.

Should we use value as the parameter of this method?

Reply:
> > Partitioned by which value
> 
> It's already defined by `Transform` of the table
> 
> I guess I know what you mean, I should change the interface parameters to `Partition`, right?

I just want to give more constraints to partition. Properties are too flexible.

Reply:
> > No exactly, these are two different attributes of a partition object, and we should not confuse them.
> 
> Should we use value as the parameter of this method?

Already have in `Partition` API's implements class

Reply:
The current implement is ok for me.